coverage/src/test/java/gov/cms/ab2d/coverage/util/CoverageDataSetup.java

```diff
@@ -0,0 +1,249 @@
+
+import gov.cms.ab2d.common.model.Contract;
+import gov.cms.ab2d.common.model.CoveragePeriod;
+import gov.cms.ab2d.common.model.Job;
+import gov.cms.ab2d.common.model.PdpClient;
+import gov.cms.ab2d.common.model.Role;
+import gov.cms.ab2d.common.repository.ContractRepository;
+import gov.cms.ab2d.common.repository.JobRepository;
+import gov.cms.ab2d.common.repository.PdpClientRepository;
+import gov.cms.ab2d.common.repository.RoleRepository;
+import gov.cms.ab2d.common.model.CoveragePeriod;
+import gov.cms.ab2d.coverage.repository.CoverageDeltaTestRepository;
+import gov.cms.ab2d.coverage.repository.CoveragePeriodRepository;
+import gov.cms.ab2d.coverage.repository.CoverageSearchEventRepository;
+import gov.cms.ab2d.coverage.repository.CoverageSearchRepository;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import javax.sql.DataSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+
+import static java.util.stream.Collectors.toList;
+
+//plan on cleaning up when decoupling contract_id foreign keys in database
+@Component
+public class CoverageDataSetup {
+
+    @Autowired
+    private DataSource dataSource;
+
+    @Autowired
+    private ContractRepository contractRepository;
+
+    @Autowired
+    private PdpClientRepository pdpClientRepository;
+
+    @Autowired
+    private JobRepository jobRepository;
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Autowired
+    private CoveragePeriodRepository coveragePeriodRepo;
+
+    @Autowired
+    private CoverageSearchRepository coverageSearchRepo;
+
+    @Autowired
+    private CoverageSearchEventRepository coverageSearchEventRepo;
+
+    @Autowired
+    CoverageDeltaTestRepository coverageDeltaTestRepository;
+
+    private final Set<Object> domainObjects = new HashSet<>();
+
+    public void queueForCleanup(Object object) {
+        domainObjects.add(object);
+    }
+
+    public void cleanup() {
+
+        // All of the coverage metadata tests assume that you completely
+        // wipe the tables between tests and that the tables started as empty tables.
+        // Based on these assumptions it is safe to simply delete everything associated
+        // with those tables
+        coverageDeltaTestRepository.deleteAll();
+        coverageDeltaTestRepository.flush();
+
+        deleteCoverage();
+
+        coverageSearchEventRepo.deleteAll();
+        coverageSearchEventRepo.flush();
+
+        coverageSearchRepo.deleteAll();
+        coverageSearchRepo.flush();
+
+        coveragePeriodRepo.deleteAll();
+        coveragePeriodRepo.flush();
+
+        List<Job> jobsToDelete = domainObjects.stream().filter(object -> object instanceof Job)
```

Comment:
Should this depend on common test for actions that are already available since most of this code is just copy-paste from the common module?

Reply:
I would say yes. Due to issues of the dependencies on the database we need to keep this. Removing it causes foriegn key errors. Once I have edited the database I can remove some of this code and remove the dependencies on common.

Reply:
I'm thinking more to extend DataSetup and then only override/implement the necessary methods.

Reply:
Yes. My plan as I decouple things more to refactor this code to not be copy paste and decoupled from common.